provisioning/miniprov/hare_mp/main.py

```diff
@@ -201,6 +202,63 @@ def test(args):
         exit(-1)
 
 
+def reset(args):
+    try:
+        rc = 0
+        consul = Consul()
+
+        keys = {'epoch': False,
+                'eq-epoch': False,
+                'last_fidk': False,
+                'leader': False,
+                'm0conf/': True,
+                'processes/': True,
+                'stats/': True}
+
+        logging.info('Deleting Hare KV entries from consul KV store')
+        for key in keys:
+            if not consul.kv.delete(key, keys[key]):
+                logging.error('Error during key delete (%s)', key)
+                exit(-1)
+
+        exit(rc)
+    except Exception as error:
+        logging.error('Error during reset (%s)', error)
+        exit(-1)
+
+
+def cleanup(args):
+    try:
+        rc = config_cleanup() & logs_cleanup()
```

Comment:
Bitwise 'and' operator doesn't have a short circuit. So even in case of an error the latter function will be executed. Is that a desired behavior?

Reply:
Done